src/jdk.compiler/share/classes/com/sun/tools/javac/file/BaseFileManager.java

```diff
@@ -267,6 +276,12 @@ public boolean handleOption(Option option, String value) {
         switch (option) {
             case ENCODING:
                 encodingName = value;
+                if ("COMPAT".equals(encodingName)) {
+                    String enc = getNativeEncoding();
+                    if (enc != null) {
+                        encodingName = enc;
+                    }
```

Comment:
If we updating javac and javadoc --encoding to support "COMPAT" then we should list this in JEP 400.

Does javadoc use doPriv already, I don't know how common it would be to run javadoc with a SM set. If the doPriv stays then you can avoid the cast by changing it to:
PrivilegedAction<String> pa = () -> System.getProperty("native.encoding");
return AccessController.doPrivileged(pa);

Reply:
javac and javadoc do not have privileged code. I am loathe to change that.